cli/src/semdep/parsers/yarn.py

```diff
@@ -235,7 +236,8 @@ def parse_yarn(
     manifest_deps = get_manifest_deps(manifest_path)
     yarn_version = 1 if lockfile_text.startswith(YARN1_PREFIX) else 2
     parser = yarn1 if yarn_version == 1 else yarn2
-    deps = safe_path_parse(lockfile_path, parser)
+    parser_name = ParserName("yarn1") if yarn_version == 1 else ParserName("yarn2")
```

Comment:
Backend needs to know what parser was used especially for cases like this where same lockfile name could be parsed differently